src/Aspire.Hosting/build/Aspire.Hosting.targets

```diff
@@ -104,14 +98,83 @@ internal static class ]]>%(ClassName)<![CDATA[
     <Warning Code="ASPIRE001" Text="The '$(Language)' language isn't fully supported by Aspire - some code generation targets will not run, so will require manual authoring." />
   </Target>
 
+  <!--
+  Gets the references in 'AppProjectTargetFramework' that aren't executable projects.
+  -->
+  <UsingTask TaskName="GetNonExecutableReferences"
+             TaskFactory="RoslynCodeTaskFactory"
+             AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <AppProjectTargetFramework ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
+      <NonExecutableReferences ParameterType="Microsoft.Build.Framework.ITaskItem[]" Output="true" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.Xml.Linq" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+          HashSet<ITaskItem> nonExecutableReferences = new HashSet<ITaskItem>();
+          
+          foreach (var appProject in AppProjectTargetFramework)
+          {
+            var additionalProperties = appProject.GetMetadata("AdditionalPropertiesFromProject");
+            if (string.IsNullOrEmpty(additionalProperties))
+            {
+              // Skip any projects that don't contain the right metadata
+              continue;
+            }
+                
+            var additionalPropertiesXml = XElement.Parse(additionalProperties);
+            foreach (var targetFrameworkElement in additionalPropertiesXml.Elements())
+            {
+              var isExe = targetFrameworkElement.Element("_IsExecutable");
+              if (isExe != null && !string.Equals(isExe.Value, "true", StringComparison.OrdinalIgnoreCase))
+              {
+                nonExecutableReferences.Add(appProject);
+              }
+            }
+          }
+
+          NonExecutableReferences = nonExecutableReferences.ToArray();
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <!--
+  Validates that all the ProjectReferences of an Aspire AppHost project are executables and
+  informs the developer to set 'AspireHostLibrary=true' if they really intended on ProjectReferencing a library.
+  -->
+  <Target Name="_ValidateAspireHostServiceReferences"
```

Comment:
While I'm totally fine with this approach, I wonder if we also evaluated the alternative approach, where the projects that require the additional metadata are the service projects, instead of the libraries. Here is my thinking: we expect most of the service project references to be added via VS, by clicking the "Enlist project to Aspire Host" item, so we can easily code that to set the additional metadata. On the other hand, it is likely that when adding a project reference to a library, this will be done manually, and therefore will today have to know to set the metadata. I wonder if we should try to optimize for having the manual case (which in this message I'm assumig it will be libraries) not require metadata.

Thoughts?

Reply:
Ok, nevermind, looks like this was already asked and discussed above. CLosing.

Reply:
Answered in https://github.com/dotnet/aspire/pull/1726#issuecomment-1905065215.